App.tsx

```diff
@@ -1,64 +1,45 @@
 import React from "react";
 import { View, Text, Button } from "react-native";
-import { useLinking, NavigationContainer } from "@react-navigation/native";
+import { NavigationContainer, LinkingOptions, useLinkTo } from "@react-navigation/native";
 import { createStackNavigator } from "@react-navigation/stack";
 import { createBottomTabNavigator } from "@react-navigation/bottom-tabs";
-import * as Linking from 'expo-linking'
-
+import * as Linking from "expo-linking";
 
 const prefix = Linking.makeUrl("/");
-const config = {
-  HomeStack: {
-    path: "stack",
-    initialRouteName: "Profile",
+const linking: LinkingOptions = {
+  prefixes: [prefix],
+  config: {
     screens: {
-      Home: "home",
-      Profile: {
-        path: "user/:id/:age",
-        parse: {
-          id: id => `there, ${id}`,
-          age: Number
-        }
-      }
-    }
+      HomeStack: {
+        path: "stack",
+        initialRouteName: "Home",
+        screens: {
+          Home: "home",
+          Profile: {
+            path: "user/:id/:age",
+            parse: {
+              id: id => `there, ${id}`,
+              age: Number,
+            },
+            stringify: {
+              id: id => id.replace("there, ", ""),
+            },
+          },
+        },
+      },
+      Settings: "settings",
+    },
   },
-  Settings: "settings"
 };
-export default function App() {
-  const ref = React.useRef();
-
-  const { getInitialState } = useLinking(ref, {
-    prefixes: [prefix],
-    config
-  });
-
-  const [isReady, setIsReady] = React.useState(false);
-  const [initialState, setInitialState] = React.useState();
-
-  React.useEffect(() => {
-    getInitialState()
-      .catch(() => {})
-      .then(state => {
-        if (state !== undefined) {
-          setInitialState(state);
-        }
-
-        setIsReady(true);
-      });
-  }, [getInitialState]);
-
-  if (!isReady) {
-    return null;
-  }
 
+export default function App() {
   function Home({ navigation }) {
```

Comment:
I think the convention is to make the functions in the main scope of the file, not inside the `App` function. It is also better for reuse. Could you change it back?

Reply:
They were already inside `App`.
I tried to make a minimal change for easier review.

Anyway, I moved them now.

Reply:
Oh, I am sorry then. But I think they shouldn't be there so it is good now.